File: backend/src/schemas/org.ts
import { z } from "zod";

export const orgCreateSchema = z.object({
  slug: z.string().min(1),
  name: z.string().min(1),
});

export const orgUpdateSchema = z.object({
  name: z.string().nullable().optional(),
  settings: z.record(z.any()).nullable().optional(),
});

export const memberRoleSchema = z.object({
  role: z.string(),
});

export const inviteSchema = z.object({
  email: z.string().email(),
  role: z.string().default("member"),
});

File: apps/api/app/schemas/org.py
from pydantic import BaseModel, EmailStr


class OrgCreateSchema(BaseModel):
    slug: str
    name: str


class OrgUpdateSchema(BaseModel):
    name: str | None = None
    settings: dict | None = None


class InviteSchema(BaseModel):
    email: EmailStr
    role: str = "member"


class OrgSummarySchema(BaseModel):
    id: str
    slug: str
    name: str
    plan: str | None
    role: str